File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr)
{
    class Stack {
        constructor() {
            this.arr = [];
        }
        
        push(item) {
            this.arr.push(item);
        }
        
        pop() {
            return this.arr.pop();
        }
        
        peek() {
            return this.arr[this.arr.length - 1];
        }
        
        size() {
            return this.arr.length;
        }
        
        isEmpty() {
            return this.arr.length === 0;
        }
    }
    var answer = [];
    
    const stack = new Stack();
    
    for(i = 0; i < arr.length; i++) {
        
        if(stack.peek() !== arr[i]) {
            stack.push(arr[i]);
        }
    }
    
    while(!stack.isEmpty()) {
        answer.push(stack.pop());
    }
    
    return answer.reverse();
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    static Stack<Integer> stack = new Stack<>();
    
    public int[] solution(int []arr) {
        for(int idx = 0; idx < arr.length; idx++) {
            if(stack.isEmpty()) {
                stack.push(arr[idx]);
            }
            else if(stack.peek() != arr[idx]) {
                stack.push(arr[idx]);
            }
        }
        
        int n = stack.size();
        int[] answer = new int[n];
        
        while(!stack.isEmpty()) {
            for(int idx = n - 1; idx >= 0; idx--) {
                answer[idx] = stack.pop();
            }
        }

        return answer;
    }
}